src/Dropdown/Dropdown.tsx

```diff
@@ -1,400 +1,160 @@
-  simpleValue: t.maybe(t.Boolean),
+  blurInputOnSelect: t.maybe(t.Boolean),
+  backspaceRemovesValue: t.maybe(t.Boolean),
   menuPosition: t.maybe(t.enums.of(["top", "bottom"])),
-  menuRenderer: t.maybe(t.Function),
-  groupByKey: t.maybe(t.String),
-  optionGroupRenderer: t.maybe(t.Function),
   placeholder: t.maybe(t.union([t.String, ReactElement])),
-  noResultsText: t.maybe(t.String),
-  allowCreate: t.maybe(t.Boolean),
-  addLabelText: t.maybe(t.String),
-  valueRenderer: t.maybe(t.Function),
-  optionRenderer: t.maybe(t.Function),
+  noOptionsMessage: t.maybe(t.String),
   delimiter: t.maybe(t.String),
   onInputChange: t.maybe(t.Function),
   onFocus: t.maybe(t.Function),
   onBlur: t.maybe(t.Function),
-  onBlurResetsInput: t.maybe(t.Boolean),
-  onCloseResetsInput: t.maybe(t.Boolean),
   isLoading: t.maybe(t.Boolean),
   id: t.maybe(t.String),
   className: t.maybe(t.String),
-  style: t.maybe(t.Object)
+  style: t.maybe(t.Object),
+  components: t.maybe(t.Object)
 };
 
-export namespace Dropdown {
-  export type SimpleValue = string | number;
-
-  export type Value = SelectNS.Option | SelectNS.Options | SimpleValue;
-
-  export type OptionRendererHandler = (
-    option: SelectNS.Option,
-    i: number
-  ) => SelectNS.HandlerRendererResult;
-
-  export type OptionGroupRendererHandler = (
-    optionGroup: string
-  ) => JSX.Element | string | null;
-
-  export interface SelectMissingMenuRendererProps {
-    // seem supported by react-select but missing in the @types file?
-    instancePrefix?: string;
-    onFocus?: () => void;
-    onOptionRef: (
-      ref: React.Component<Dropdown.Props> | null,
-      isFocused: boolean
-    ) => void;
-    onSelect?: () => void;
-    optionClassName?: string;
-    optionComponent: React.ComponentType<{ [k: string]: any }>; // TODO
-    optionRenderer: OptionRendererHandler;
-    optionGroupRenderer: OptionGroupRendererHandler;
-    valueKey: string;
-  }
-  export interface CustomMenuRendererProps {
-    groupByKey?: string;
-  }
-  export type MenuRendererProps = SelectNS.MenuRendererProps &
-    SelectMissingMenuRendererProps &
-    CustomMenuRendererProps;
-  export type MenuRendererHandler = (
-    props: MenuRendererProps
-  ) => JSX.Element[] | SelectNS.HandlerRendererResult;
-
-  export type MenuPosition = "top" | "bottom";
-
-  export type Props = RequiredProps & Partial<DefaultProps>;
-}
-
-type DefaultedProps = RequiredProps & DefaultProps;
+const defaultComponents: Dropdown.Props<any>["components"] = {
+  Placeholder: ({ children }) => (
+    <div className="react-select__placeholder">{children}</div>
+  ),
+  IndicatorSeparator: () => null,
+  Option: props => (
+    <components.Option
+      {...props}
+      className={cx({
+        "is-disabled": props.isDisabled,
+        "is-focused": props.isFocused,
+        "is-selected": props.isSelected
+      })}
+    />
+  ),
+  DropdownIndicator: props => (
+    <span className="react-select__select-arrow-zone">
```

Comment:
same: if we have to re-define them then I would implement a simple `Arrow` component instead of using their `className` and relying on their layout